### PR TITLE
Fix pool ops

### PR DESCRIPTION
**Pull request by fumihwh**
Rewrite `_compatibility_pool ` method.

### CODE REVIEW

**Comment by tjingrant** [comment]
Hi, I actually wrote the pooling function just to cover the case of average pooling with `SAME_UPPER` auto pad attribute. Max pooling with SAME_UPPER auto pad attribute does not need such complications because you can use tf.pad and then perform the pooling with `VALID` padding.

**Comment by fumihwh** [comment]
`tf.pad` will pad 0 which will get unexpected result if all numbers in window is negative.

**Comment by tjingrant** [comment]
@fumihwh to be fair, you can pad a very negative number...

**Comment by tjingrant** [comment]
@fumihwh let's keep your solution since it's more logical (and consistent with avg pool) than tf.pad with a very negative number.

**Comment by fumihwh** [comment]
@tjingrant Oh...maybe you misunderstood.
Just auto_pad=SAME_LOWER or None will call `_compatibility_pool`. 
`SAME_UPPER` and `VALID` call `tf.nn.pool`.

**Comment by tjingrant** [comment]
Sorry my typo, in max_pooling with SAME_LOWER, we apply the explicit padding using a very negative number, something like -9999.999 (or FLOAT_MIN to be more accurate) and expect the results to be like real pads in max_pooling op.

**Comment by tjingrant** [comment]
Python 2 keeps having trouble with vgg16 memory allocation. Trying a 3rd time.

**Comment by tjingrant** [comment]
I'm fine with merging since python 3 works and previous logs suggest that vgg16 was the one to blame.

**Comment by tjingrant** [comment]
Great, succeeded on the 4th trial. Let's hope it won't happen again.
